src/ucp/wireup/wireup_cm.c

```diff
@@ -494,16 +494,16 @@ static unsigned ucp_ep_cm_disconnect_progress(void *arg)
     ucs_trace("ep %p: got remote disconnect, cm_ep %p", ucp_ep, uct_cm_ep);
     ucs_assert(ucp_ep_get_cm_uct_ep(ucp_ep) == uct_cm_ep);
 
-    ucp_ep_invoke_err_cb(ucp_ep, UCS_ERR_CONNECTION_RESET);
```

Comment:
so if (ucp_ep->flags & UCP_EP_FLAG_LOCAL_CONNECTED) is false, user err callback is not called?

Reply:
yes, according to protocol flow the EP is closed by user in this case. I added corresponding assert.

Reply:
you mean that this is the flow of user closing the EP, and before this PR we called user err handler, but really we shouldn't?
can you pls add code comment here for this?

Reply:
> you mean that this is the flow of user closing the EP

yes

> before this PR we called user err handler

no,  `ucp_ep_invoke_err_cb` has check on flag, but the ep can be closed immediately from the CB